### ISSUE TITLE
SinricPro handler blocked by millis() function (or Blink Without Delay concept) in loop()

### ISSUE THREAD

**Issue by mareskav** [sinricpro/esp8266-esp32-sdk#267]
Hi there!

Could I ask you for help please? I ran into a problem where I am not able to run `SinricPro.handle()` together with `millis()` and **Blink Without Delay** example (https://www.arduino.cc/en/Tutorial/BuiltInExamples/BlinkWithoutDelay). It looks like that  `millis() ` function **is too much blocking SinricPro task in loop()**, so it stops to response to my switch button ON / OFF clicks.

**My goal** is to keep running ESP 8266 for an hour (before  `ESP.deepSleep(fooTime)` will be called) and enjoy SinricPro websocket communication with Google Home. I saw a few examples of non-blocking codes here in issues, but couldn´t find anything what could solve my problem. Is it maybe job for _Cooperative Multitasking_? 🤔 

I went through a few code iterations, so not posting the code, but I can show an example if it would be helpfull of course. **Basically, I just want**:
- Run ESP 8266 with responsive  `SinricPro.handle()` switch and "non-blocking delay" for 1 hour or more ❓ 
- Deep sleep for some time (this is easy)

Btw, thank you for the great work with SinricPro library! It´s amazing and soo easy to use! ❤️

**Comment by sivar2311** [comment]
Hi @mareskav!

Neither `millis()` nor `SinricPro.handle()` block the sketch.

Basically, `SinricPro.handle()` is responsible for processing incoming requests and calling the corresponding callback functions of the devices. It is also responsible for sending the events of the devices to the SinricPro server.

I could only think of two scenarios that would lead to blocking:

- Your callback functions contain blocking code.
- Your sketch sends excessive events. In this case an artificial delay takes place to protect the SinricPro server from flooding. This is made visible with a corresponding serial output.

In general, your project can be implemented without any problems.

**Comment by mareskav** [comment]
Thank you for quick response @sivar2311 ! And sorry for my slow one.

I tested several iterations of my code and I concluded that:
- `millis()` function is not blocking the code, maybe just slowing the code a little bit down, that´s all.
- Now, I am able to use `NTPClient` (https://github.com/arduino-libraries/NTPClient) to effectively schedule tasks.
- I was stupid and using AccelStepper (https://github.com/waspinator/AccelStepper) library with function handling step motor moves, but blocking the code.

Thank you for hints! I guess I should take care of "Your sketch sends excessive events. In this case an artificial delay takes place to protect the SinricPro server from flooding. This is made visible with a corresponding serial output." Especially when I would decide to handle more devices at once in short time.
